pulp_rpm/app/tasks/prune.py

```diff
@@ -35,17 +38,33 @@ def prune_repo_packages(repo_pk, keep_days, dry_run):
     log.debug(f">>> TOTAL RPMS: {curr_vers.get_content(Package.objects).count()}")
 
     # We only care about RPM-Names that have more than one EVRA - "singles" are always kept.
-    rpm_by_name_age = (
-        curr_vers.get_content(Package.objects.with_age())
-        .filter(age__gt=1)
-        .order_by("name", "epoch", "version", "release", "arch")
-        .values("pk")
+    rpm_by_name = (
+        curr_vers.get_content(Package.objects)
+        .order_by(
+            "name", "epoch", "version", "release", "arch"
+        )  # why sort by all of these values?  lexicographical sort is not "accurate"
```

Comment:
I have been doing experiments with the "CREATE COLLATION" feature in postgresql.
I think it could be tailored to define a proper "lexicographical" order for these version components.

related pr: https://github.com/pulp/pulp_deb/pull/1335

Reply:
RPM versioning has a special `~` and `^` character behavior - is collation flexible enough to handle that?

see: 

https://slashterix.wordpress.com/2016/08/06/rpm-version-comparison/

https://github.com/pulp/pulp_rpm/blob/main/pulp_rpm/app/rpm_version.py

Reply:
> I have been doing experiments with the "CREATE COLLATION" feature in postgresql. I think it could be tailored to define a proper "lexicographical" order for these version components.
> 
> related pr: [pulp/pulp_deb#1335](https://github.com/pulp/pulp_deb/pull/1335)

Isn't that also dependent on pgres-16?

Reply:
Yes it is pg 16 and for the special ~ handling (sorting even before the end of the segment) i needed to trick a little.
I'd love to do a little less of that. But i found no way to either add a custom normalize function to the collation or a way to provide a search_key function to a text field.